src/mesh.c

```
@@ -6,381 +6,360 @@
+            /* Decrypt handshake payload using network key first to authenticate source */
+            uint8_t plain[sizeof(handshake_message_t)];
+            secure_nonce_t nonce;
+            memset(&nonce, 0, sizeof(nonce));
+            memcpy(&nonce.value, &packet->header.nonce_value, 8);
+
+            int dec = security_decrypt(mesh_state.network_key, &nonce,
+                                       packet->payload, packet->payload_len,
+                                       (uint8_t*)&packet->header, sizeof(packet_header_t),
+                                       packet->tag, plain);
+
+            if (dec < 0 || (size_t)dec < sizeof(handshake_message_t)) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)